File: frontend/src/api/aliases.ts
import api from './index'

export interface MerchantAlias {
  id: number
  family_id: number | null
  original_name: string
  alias_name: string
  category_id: number | null
  sub_category_id: number | null
  platform_id: number | null
  hit_count: number
  is_active: boolean
}

export interface AliasCreate {
  original_name: string
  alias_name: string
  category_id?: number
  sub_category_id?: number
  platform_id?: number
}

export interface AliasUpdate {
  alias_name?: string
  category_id?: number
  sub_category_id?: number
  platform_id?: number
}

export function getAliases() {
  return api.get<MerchantAlias[]>('/aliases')
}

export function createAlias(data: AliasCreate) {
  return api.post<MerchantAlias>('/aliases', data)
}

export function updateAlias(aliasId: number, data: AliasUpdate) {
  return api.put<MerchantAlias>(`/aliases/${aliasId}`, data)
}

export function deleteAlias(aliasId: number) {
  return api.delete(`/aliases/${aliasId}`)
}

File: backend/app/api/aliases.py
import structlog
from fastapi import APIRouter, Depends, HTTPException, status
from sqlalchemy import select
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.deps import get_current_user
from app.database import get_db
from app.models.merchant_alias import MerchantAlias
from app.models.user import User
from app.schemas.extra import AliasCreate, AliasUpdate, AliasResponse

logger = structlog.get_logger()
router = APIRouter(tags=["商户别名"])


@router.get("/api/aliases", response_model=list[AliasResponse])
async def list_aliases(
    current_user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    result = await db.execute(
        select(MerchantAlias)
        .where(MerchantAlias.family_id == current_user.family_id)
        .order_by(MerchantAlias.hit_count.desc())
    )
    return [AliasResponse.model_validate(a) for a in result.scalars()]


@router.post("/api/aliases", response_model=AliasResponse, status_code=201)
async def create_alias(
    body: AliasCreate,
    current_user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    alias = MerchantAlias(family_id=current_user.family_id, **body.model_dump())
    db.add(alias)
    await db.commit()
    await db.refresh(alias)
    return AliasResponse.model_validate(alias)


@router.put("/api/aliases/{alias_id}", response_model=AliasResponse)
async def update_alias(
    alias_id: int,
    body: AliasUpdate,
    current_user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    result = await db.execute(
        select(MerchantAlias).where(
            MerchantAlias.id == alias_id,
            MerchantAlias.family_id == current_user.family_id,
        )
    )
    alias = result.scalar_one_or_none()
    if not alias:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="别名不存在")
    for field, value in body.model_dump(exclude_unset=True).items():
        setattr(alias, field, value)
    await db.commit()
    await db.refresh(alias)
    return AliasResponse.model_validate(alias)


@router.delete("/api/aliases/{alias_id}", status_code=204)
async def delete_alias(
    alias_id: int,
    current_user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    result = await db.execute(
        select(MerchantAlias).where(
            MerchantAlias.id == alias_id,
            MerchantAlias.family_id == current_user.family_id,
        )
    )
    alias = result.scalar_one_or_none()
    if not alias:
        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="别名不存在")
    await db.delete(alias)
    await db.commit()
